### PR TITLE
Enable TS linting and add a git hook script for pre-push linting; lint cleanup.

### PR DESCRIPTION
**Pull request by markfickett**
As in the README, run this to set up the pre-push hook:
```
mkdir -p .git/hooks
cat > .git/hooks/pre-push <<EOF
#!/bin/bash -e
git-hooks/pre-push
EOF
```

### CODE REVIEW

**Comment by markfickett** [comment]
@dmohs PR for you.

File: README.asciidoc [hunk 1]
```diff
@@ -15,6 +15,14 @@ Install project deps:
...
  npm install
 
+Add git hooks:
+
+ mkdir -p .git/hooks
+ cat > .git/hooks/pre-push <<EOF
+ #!/bin/bash -e
+ git-hooks/pre-push
+ EOF
+
```
Comment (dmohs) [inline README.asciidoc]: This is the only part of this that I don't love—that each developer has to remember to run this. Perhaps it is more trouble than it is worth, but we could use something like this:
https://github.com/kilianc/shared-git-hooks
to enforce that this step isn't forgotten.

**Review by dmohs — APPROVED** [review]
Love this. One comment/suggestion. Use your judgement—I'm happy to merge as-is.

File: README.asciidoc (hunk 1 above)
Comment (markfickett) [reply README.asciidoc]: Awesome! Switched to using that.

**Comment by markfickett** [comment]
Ready for merge @dmohs .

**Comment by dmohs** [comment]
Nice!

**Pull request merged**